prov/tcp/src/tcpx_progress.c

```diff
@@ -641,6 +641,18 @@ void tcpx_progress_tx(struct tcpx_ep *ep)
 		tx_entry = container_of(entry, struct tcpx_xfer_entry, entry);
 		process_tx_entry(tx_entry);
 	}
+	bool is_shutdown = false;
+
+	if (ep->cm_state == TCPX_EP_SHUTDOWN && !ep->poll_released)
+		is_shutdown = true; /* let it progress once more */
+
+
+	if (is_shutdown)
+	{
+		tcpx_ep_wait_fd_del(ep);
+		ep->poll_released = true; /* just an optimization */
+	}
+
```

Comment:
There's extra spacing in this change, and the code could be simplified as:
```
if (ep->cm_state == TCPX_EP_SHUTDOWN && !ep->poll_released) {
    tcpx_ep_wait_fd_del(ep);
    ep->poll_released = true;
}
```
That said, I'd like to avoid have these checks as part of every progress call.  The cleanup should occur wherever the state is being set to SHUTDOWN, and ideally, the state should indicate if the ep is part of the wait_fd or not.  I.e. we shouldn't need yet another state value (poll_released) to track this.